docs/distribution/README.md

```diff
@@ -0,0 +1,923 @@
+# OCI Distribution - Notary v2 Signature Support
+
+To support [Notary v2 goals][notaryv2-goals], upload, persistance and discovery of signatures must be supported.
+
+To minimize the complexity of registry operators and projects to adopt Notary v2, a balance between leveraging what already exists and new patterns to support secure discovery are explored.
+
+## Table of Contents
+
+* [Signature Persistance](#signature-persistance)
+* [Signature Push](#signature-push)
+* [Signature Discovery](#signature-discovery)
+* [Signature Pull](#signature-pull)
+* [Example Artifacts](#example-artifacts)
+
+## Signature Persistance
+
+Several options for how to persist a signature were explored. We measure these options against the [goals of Notary v2][notaryv2-goals], specifically:
+
+* Maintain the original artifact digest and collection of associated tags, supporting existing dev through deployment workflows
+* Multiple signatures per artifact, enabling the originating vendor signature, public registry certification and user/environment signatures
+* Native persistance within an OCI Artifact enabled, distribution*spec based registry
+* Artifact and signature copying within and across OCI Artifact enabled, distribution spec based registries
```

Comment:
Now that dist-spec is brought into this discussion, is dist-spec a NECESSARY condition? or for example, can docker registry be notary-v2 compliant without being dist-spec compliant?

Reply:
We don't believe we can achieve the e2e goals of Notary v2 without them. If you consider Notary v1/Docker Content Trust required a significant additional service, which had a host of additional issues, we believe this is a reasonable bar.
For instance, our design goals require linking a signature with an artifact without changing the original artifacts tag or digest. This requires another artifact to reference it. We looked into one manifest referring to another with content in the config object. But, that would require registries to implement ref counting, by parsing the `manifest.config` to support garbage collection.
We're now focusing on using Index as it has a linking mechanism. In this case, we need a way to differentiate multi-arch container inedexes, from Notary v2 and CNAB based indexes.
Lastly, we need an api to find the signatures, based on a target artifact.
All of these require changes. The index changes focus on leveraging the hardest parts of registry implementations (ref counting & garbage collection) while adding capabilities that have broader needs.
We'll also need changes to existing clients to pull the workflows together.
We hope the value will be enough, that the relatively small community of people that own the impacted products would be motivated to add the relatively minimal amount of work to enable the e2e workflows. But, that will ultimately be the test of the effort - adoption.

Was there a specific reason you were concerned about changes?

Reply:
re-reading the question, I think we're talking more about compliance. We've been capturing registry compliance [here](https://github.com/opencontainers/oci-conformance/tree/master/distribution-spec)